### PR TITLE
EA Origin presence

### PR DESCRIPTION
**Pull request by CoopJax**
This is basic, as I have a basic knowledge of typescript. But I hope it looks good.

### CODE REVIEW

File: websites/O/Origin/dist/metadata.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  },
+  "service": "EA Origin",
+  "description": {
+    "en": "Play great PC games and connect with your friends, all in one place."
+  },
+  "url": "origin.com",
+  "version": "10.4.74-2461",
+  "logo": "https://i.imgur.com/M5aQ6b2.png",
+  "thumbnail": "https://i.imgur.com/M5aQ6b2.png",
+  "color": "#F56C2D",
```
Comment (Bas950) [inline websites/O/Origin/dist/metadata.json:14]: thumbnail should be a promotional image or a screenshot of the website frontpage.

File: websites/O/Origin/dist/metadata.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    "name": "web",
+    "id": "526087204674600966"
+  },
+  "service": "EA Origin",
+  "description": {
+    "en": "Play great PC games and connect with your friends, all in one place."
+  },
+  "url": "origin.com",
+  "version": "10.4.74-2461",
+  "logo": "https://i.imgur.com/M5aQ6b2.png",
```
Comment (Bas950) [inline websites/O/Origin/dist/metadata.json:12]: version 1.0.0

File: websites/O/Origin/presence.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
+var presence = new Presence({
+    clientId: "760611329588985886"
+  }),
+  presenceData: PresenceData = {
+    largeImageKey: "icon"
+  },
+  customData = false;
+
+presence.on("UpdateData", async () => {
+  customData = false;
```
Comment (Bas950) [inline websites/O/Origin/presence.ts:10]: what is this custom data stuff?
Because its never set to true...

File: websites/O/Origin/presence.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    };
+    if (document.location.pathname.startsWith("en-us/store")) {
+    presenceData.details = "Browsing for a game";
+  } else if (document.location.pathname.startsWith("/en-us/search")) {
+    presenceData.details = "Searching the store";
+  } else if (document.location.pathname.startsWith("/en-us/profile/")) {
+    presenceData.details = "Viewing a user's profile";
+  } else if (document.location.pathname.startsWith("/en-us/profile")) {
+    presenceData.details = "Looking through their friends list.";
+  } else if (document.location.pathname.startsWith("en-us/friends")) {
```
Comment (Bas950) [inline websites/O/Origin/presence.ts:25]: add support for all pages, and no matter what language

File: websites/O/Origin/presence.ts [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    presenceData.details = "404 Page Not Found...";
+  }
+  if (!customData) {
+    presence.setActivity(presenceData);
+  }
+});
+
+presence.on("iFrameData", function (data) {
+  console.log(data);
+});
```
Comment (Bas950) [inline websites/O/Origin/presence.ts:38]: Why is this in here when you have no iframe?

**Review by Bas950 — CHANGES_REQUESTED** [review]
Upload proof of it working

File: websites/O/Origin/dist/metadata.json (hunk 1 above)
Comment (CoopJax) [reply websites/O/Origin/dist/metadata.json:14]: ok

File: websites/O/Origin/dist/metadata.json (hunk 2 above)
Comment (CoopJax) [reply websites/O/Origin/dist/metadata.json:12]: So it's the plugin version not the site ver?

File: websites/O/Origin/presence.ts (hunk 4 above)
Comment (CoopJax) [reply websites/O/Origin/presence.ts]: It should support all languages now.

**Comment by CoopJax** [comment]
I can't figure out how to get it working

**Comment by CoopJax** [comment]
Does it show up on the plugin list?

File: websites/O/Origin/presence.ts [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+  } else if (document.location.pathname.startsWith("/search")) {
+    presenceData.details = "Searching the store";
+  } else if (document.location.pathname.startsWith("/profile/")) {
+    presenceData.details = "Viewing a friend's profile";
+  } else if (document.location.pathname.startsWith("/profile")) {
+    presenceData.details = "Looking through their friends list.";
+  } else if (document.location.pathname.startsWith("profile/friends")) {
+    };
+    presence.setActivity(profileData);
+  } else {
```
Comment (Bas950) [inline websites/O/Origin/presence.ts:27]: this seems like 1 big error...

**Review by Bas950 — CHANGES_REQUESTED** [review]
and there are way more pages to support...

File: websites/O/Origin/presence.ts (hunk 6 above)
Comment (CoopJax) [reply websites/O/Origin/presence.ts:27]: hmm...

**Comment by CoopJax** [comment]
I added library and wishlist support. I should add achievements though...

File: websites/O/Origin/presence.ts (hunk 6 above)
Comment (CoopJax) [reply websites/O/Origin/presence.ts:35]: what should I do to change it?

**Comment by CoopJax** [comment]
Well something went wrong here

File: websites/O/Origin/presence.ts [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+
+presence.on("UpdateData", async () => {
+
+  if (document.location.pathname == "/my-home") {
+    presenceData.details = "Browsing the homepage";
+  } else if (document.location.pathname.startsWith("/store/download")) {
+    presenceData.details = "Installing the Origin launcher";
+    };
+    if (document.location.pathname.startsWith("/store")) {
+    presenceData.details = "Browsing for a game";
```
Comment (Bas950) [inline websites/O/Origin/presence.ts:17]: you should also make it like changing, when you go to a games store page, it shows the name of the game etc.

File: websites/O/Origin/presence.ts [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+  } else if (document.location.pathname.startsWith("/profile/achievements")) {
+    presenceData.details = "Viewing their achievement progress";
+  } else if (document.location.pathname.startsWith("/eaAccounts")) {
+    presenceData.details = "Editing their settings";
+  } else if (document.location.pathname.startsWith("/game-library")) {
+    presenceData.details = "Browsing their game library";
+  } else if (document.location.pathname.startsWith("/search")) {
+    presenceData.details = "Searching the store";
+  } else if (document.location.pathname.startsWith("/profile")) {
+    presenceData.details = "Viewing a profile";
```
Comment (Bas950) [inline websites/O/Origin/presence.ts:27]: show which, IE name of user

**Review by Bas950 — CHANGES_REQUESTED** [review]
and there are way more pages to support...

File: websites/O/Origin/presence.ts (hunk 8 above)
Comment (CoopJax) [reply websites/O/Origin/presence.ts:27]: How?

Comment (ririxichan) [reply websites/O/Origin/presence.ts:27]: Get for example from url?

Comment (Bas950) [reply websites/O/Origin/presence.ts:27]: read elements from page...

**Comment by CoopJax** [comment]
I added more languages to support! I hope that that's better.

**Comment by ririxichan** [comment]
image
fix. (you should use lang codes like en, pl not english etc.

**Comment by CoopJax** [comment]
> 
> 
> image
> fix. (you should use lang codes like en, pl not english etc.
> also fix presence too
> image
> there should be `}` after this

so en, de, es,  etc?

**Comment by CoopJax** [comment]
👍🏻

**Comment by CoopJax** [comment]
NOOO IT FAILED AGAIN :(

File: websites/O/Origin/dist/metadata.json [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+    "zh": "遊玩優質 PC 遊戲並與好友聯繫，一個地方達成所有事。"
+  },
+  "service": "Origin",
+  "version": "1.0.0",
+  "logo": "https://i.imgur.com/M5aQ6b2.png",
+  "thumbnail": "https://i.imgur.com/Ov7QVpj.jpg",
+  "color": "#F56C2D",
+  "tags": [
+    "Social",
+    "Games"
```
Comment (ririxichan) [inline websites/O/Origin/dist/metadata.json:34]: first letter shouldn't be capitalized.

**Comment by CoopJax** [comment]
bruh bot spam

**Review by ririxichan — CHANGES_REQUESTED** [review]
still fix presence and check bas things that you should do.

File: websites/O/Origin/dist/metadata.json (hunk 9 above)
Comment (CoopJax) [reply websites/O/Origin/dist/metadata.json:34]: oh ok

File: websites/O/Origin/dist/metadata.json [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+    "jp": "名作PCゲーム。フレンドとのつながり。 その全てはここに。",
+    "ko": "명작 PC 게임, 친구들과의 소통. 한 곳에서 모두 즐기세요!",
+    "nl": "Speel geweldige PC-games en maak verbinding met je vrienden, allemaal op één plek.",
+    "no": "Spill toppspill for PC og kom i kontakt med vennene dine – alt på ett sted.",
+    "pl": "Graj we wspaniałe gry na PC i kontaktuj się ze znajomymi — wszystko w jednym miejscu.",
+    "pt": "Jogue grandes títulos para PC e conecte-se com seus amigos, tudo em um só lugar.",
+    "ru": "Играйте в любимые игры для ПК и общайтесь с друзьями в одном месте.",
+    "sv": "Spela PC-spelen du älskar och umgås med dina vänner, allt på ett enda ställe.",
+    "th": "เล่นเกม PC ชั้นยอดต่างๆ และเชื่อมต่อกับเพื่อนๆ ของคุณได้ในที่เดียว",
+    "zh": "遊玩優質 PC 遊戲並與好友聯繫，一個地方達成所有事。"
```
Comment (Bas950) [inline websites/O/Origin/dist/metadata.json:25]: some of the ids arent correct, use these: https://api.premid.app/v2/langFile/list

**Review by Bas950 — CHANGES_REQUESTED** [review]

File: websites/O/Origin/dist/metadata.json (hunk 10 above)
Comment (CoopJax) [reply websites/O/Origin/dist/metadata.json:25]: should I do it like JA_JP instead of jp?

Comment (CoopJax) [reply websites/O/Origin/dist/metadata.json]: I should have gotten everything...

**Comment by CoopJax** [comment]
How do I get this to display?

File: websites/O/Origin/presence.ts (hunk 7 above)
Comment (CoopJax) [reply websites/O/Origin/presence.ts:14]: ok

**Comment by CoopJax** [comment]
I _think_ that i got the custom game/username working.

File: websites/O/Origin/presence.ts (hunk 7 above)
Comment (CoopJax) [reply websites/O/Origin/presence.ts:14]: done...

File: websites/O/Origin/dist/metadata.json [hunk 11]
```diff
@@ -13,16 +13,16 @@
     "fi": "Pelaa upeita PC-pelejä ja pidä yhteyttä ystäviisi yhdessä ja samassa paikassa.",
     "fr": "Jouez à de grands jeux PC et connectez-vous avec vos amis sur une seule et même plateforme.",
     "it": "Gioca ai migliori titoli PC e resta in contatto con i tuoi amici, il tutto in un unico spazio.",
-    "jp": "名作PCゲーム。フレンドとのつながり。 その全てはここに。",
+    "ja_JP": "名作PCゲーム。フレンドとのつながり。 その全てはここに。",
     "ko": "명작 PC 게임, 친구들과의 소통. 한 곳에서 모두 즐기세요!",
     "nl": "Speel geweldige PC-games en maak verbinding met je vrienden, allemaal op één plek.",
```
Comment (Bas950) [inline websites/O/Origin/dist/metadata.json]: pretty sure that has to be ko_KR, so double check everything again mate xD

**Review by Bas950 — CHANGES_REQUESTED** [review]

File: websites/O/Origin/dist/metadata.json (hunk 11 above)
Comment (CoopJax) [reply websites/O/Origin/dist/metadata.json]: oof - i'll get to it

**Comment by CoopJax** [comment]
I think I broke codacy with this last commit...

File: websites/O/Origin/presence.ts [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+  } else if (document.location.pathname.startsWith("profile/friends")) {
+    presenceData.details = "Viewing their friends list";
+  };
+  var profileData: PresenceData = {
+      details: "Viewing " + name + "'s Profile",
+      largeImageKey: "logo"
+    };
+    
+  var storeData: PresenceData = {
+      details: "Viewing " + game ,
```
Comment (ririxichan) [inline websites/O/Origin/presence.ts:38]: where you get variable ``name`` and ``game``? Because it isn't defined at all anywhere.

**Review by ririxichan — CHANGES_REQUESTED** [review]

File: websites/O/Origin/presence.ts [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+    presenceData.details = "Viewing their wishlist";
+  } else if (document.location.pathname.startsWith("profile/friends")) {
+    presenceData.details = "Viewing their friends list";
+  };
+  var profileData: PresenceData = {
+      details: "Viewing " + name + "'s Profile",
+      largeImageKey: "logo"
+    };
+    
+  var storeData: PresenceData = {
```
Comment (Bas950) [inline websites/O/Origin/presence.ts:37]: no var, and then you can merge these consts

File: websites/O/Origin/presence.ts [hunk 14]
```diff
@@ -0,0 +1,50 @@
...
+    largeImageKey: "icon"
+  };
+
+presence.on("UpdateData", async () => {
+  if (document.location.pathname == "/my-home") {
+    presenceData.details = "Browsing the homepage";
+  } else if (document.location.pathname.startsWith("/store/download")) {
+    presenceData.details = "Installing the Origin launcher";
+    if (document.location.pathname.startsWith("/store")) {
+    presenceData.details = "Browsing for a game";
```
Comment (Bas950) [inline websites/O/Origin/presence.ts:14]: this if doesnt get closed..

**Review by Bas950 — CHANGES_REQUESTED** [review]
and there are way more pages to support...

File: websites/O/Origin/presence.ts (hunk 14 above)
Comment (CoopJax) [reply websites/O/Origin/presence.ts]: Ok, I fixed it.

File: websites/O/Origin/presence.ts (hunk 13 above)
Comment (CoopJax) [reply websites/O/Origin/presence.ts:37]: ? I'm confused.

Comment (Bas950) [reply websites/O/Origin/presence.ts:37]: You shouldnt use var, you should use let or const, in this case const, and in the case you can also merge the consts

const var1 = obj,
var2 = obj2;

Comment (CoopJax) [reply websites/O/Origin/presence.ts:37]: oh ok 👍🏻

Comment (CoopJax) [reply websites/O/Origin/presence.ts:37]: Yeah, iv'e hit a wall.

Comment (CoopJax) [reply websites/O/Origin/presence.ts:37]: it's const var {
details blah blah
imagekey
} right?

Comment (CoopJax) [reply websites/O/Origin/presence.ts:37]: or am I not getting this at all xD

**Comment by Bas950** [comment]
I suggest you make a new PR, when you have added some better statuses and learned a bit more about how TS/JS works.

**Pull request closed without merging**